lms/djangoapps/teams/models.py

```diff
@@ -1,18 +1,74 @@
 """Django models related to teams functionality."""
 
+from datetime import datetime
 from uuid import uuid4
 import pytz
 from datetime import datetime
 
+from django.core.exceptions import ObjectDoesNotExist
 from django.contrib.auth.models import User
 from django.db import models
+from django.dispatch import receiver
 from django.utils.translation import ugettext_lazy
 from django_countries.fields import CountryField
 
+from django_comment_common.signals import (
+    thread_created,
+    thread_edited,
+    thread_deleted,
+    thread_voted,
+    comment_created,
+    comment_edited,
+    comment_deleted,
+    comment_voted,
+    comment_endorsed
+)
 from xmodule_django.models import CourseKeyField
 from util.model_utils import generate_unique_readable_id
 from student.models import LanguageField, CourseEnrollment
 from .errors import AlreadyOnTeamInCourse, NotEnrolledInCourseForTeam
+from teams import TEAM_DISCUSSION_CONTEXT
+
+
+@receiver(thread_voted)
+@receiver(thread_created)
+@receiver(comment_voted)
+@receiver(comment_created)
+def post_create_vote_handler(sender, **kwargs):  # pylint: disable=unused-argument
+    """Receive user activity signals from django_comment_client and
+    discussion_api and update the user's last activity date, if the
+    discussion in which activity took place has the team context.
+    """
+    post = kwargs['post']
+    if getattr(post, "context", "course") == TEAM_DISCUSSION_CONTEXT:
+        CourseTeamMembership.update_last_activity(kwargs['user'], post.commentable_id)
+
+
+@receiver(thread_edited)
+@receiver(thread_deleted)
+@receiver(comment_edited)
+@receiver(comment_deleted)
+def post_edit_delete_handler(sender, **kwargs):  # pylint: disable=unused-argument
```

Comment:
Can you create a helper method for these 3 repeated blocks of code (with a boolean to indicate whether the user who performed the action must also be original author?